api/v1alpha1/tls_types.go

```diff
@@ -0,0 +1,59 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+type TLSSettings struct {
+	// Version details the minimum/maximum TLS protocol version that
+	// should be supported by this listener.
+	// +optional
+	Version *TLSVersions `json:"version,omitempty"`
+
+	// CipherSuites specifies the set of cipher suites supported when
+	// negotiating TLS 1.0 - 1.2. This setting has no effect for TLS 1.3.
+	// +optional
+	CipherSuites []string `json:"ciphers,omitempty"`
+
+	// ECDHCurves specifies the set of supported ECDH curves.
+	// +optional
+	ECDHCurves []string `json:"ecdhCurves,omitempty"`
+
+	// SignatureAlgorithms specifies which signature algorithms the listener should
+	// support.
+	// +optional
+	SignatureAlgorithms []string `json:"signatureAlgorithms,omitempty"`
```

Comment:
Also, are there any use cases that need to configure CipherSuites,ECDHCurves, and SignatureAlgorithms?

Reply:
If TLS protocol versions below 1.2 are used, the default cipher suite is not going to work. See the documentation [here](https://www.envoyproxy.io/docs/envoy/v1.28.0/api-v3/extensions/transport_sockets/tls/v3/common.proto#extensions-transport-sockets-tls-v3-tlsparameters).

The ability to control the supported curves and signature algorithms is needed for some scenarios where additional hardening is needed - the customer wants to specify exactly what should be used.

While most use-cases shouldn't need to touch the defaults, sometimes it's required.

Reply:
defer to @envoyproxy/gateway-maintainers for more inputs.

Reply:
(This is a rare use case, but it is a real use case.)

Reply:
does this need to be an enum ?

Reply:
This field is so much of a moving target that even the [Envoy Proxy documentation](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/transport_sockets/tls/v3/common.proto) doesn't specify what the up-to-date values are. I'm not sure it makes sense to have an enum here.

Reply:
cool, lets keep it as is